lib/api/src/identity/extensions/Identity.ts

```diff
@@ -0,0 +1,20 @@
+import { Identity as AbstractIdentity } from '../Identity'
+
+export class Identity extends AbstractIdentity {
+
+  /**
+   * {@inheritdoc}
+   */
+  public getRedirectUrl (path: string) : string {
+    return `${browser.identity.getRedirectURL()}${path}`
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public launchAuthFlow (url: string, interactive: boolean = true) : Promise<string> {
+    return browser.identity.launchWebAuthFlow({ url, interactive })
+      .catch(error => Promise.reject(error))
```

Comment:
An alternative would be `.catch(Promise.reject)`.

Reply:
much nicer 😄

Reply:
We would want to make a custom exception though, to make it consistent for extensions/chrome.

Reply:
👍